### PR TITLE
Include all files in the sdist via MANIFEST.in

### LINKED ISSUE #414: get rid of MANIFEST or automate its maintenance
we keep having issues with maintaining a good MANIFEST.in. in https://github.com/linkchecker/linkchecker/pull/410#issuecomment-633222999, @cjmayo estimates that simply getting rid of it would only add 10KB to the source distribution.

so let's just get rid of it? or what purpose does it serve? i'm not particularly familiar with it, to be honest...

### PR DESCRIPTION
**Pull request by mgedmin**
This PR

- gets rid of the `MANIFEST` file generated by custom build commands in setup.py and the Makefile
- updates MANIFEST.in to include ALL THE FILES!
- adds check-manifest to CI so we can be sure MANIFEST.in will always include ALL THE FILES!

Closes #414.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,30 +1,93 @@
...
-include scripts/*.sh scripts/*.py
-# tests
-include robots.txt
-include tests/configuration/data/*.ini
-recursive-include tests *.py *.result *.html *.ico *.txt *.zip *.asc *.css *.xhtml *.sqlite *.adr *.swf *.bin *.markdown *.pdf *.php *.plist *.wml *.xml
+include install-rpm.sh
+include linkchecker.freecode
+include .project
+include .pydevproject
+  doc/web/wokconfig
```
Comment (cjmayo) [inline MANIFEST.in:13]: Don't think it matters for linkchecker but N.B. something went wrong here.

warning: manifest_maker: MANIFEST.in, line 13: unknown action 'doc/web/wokconfig'

File: MANIFEST.in [hunk 2]
```diff
@@ -1,30 +1,93 @@
...
+    *.ico                               \
+    *.jpg                               \
+    *.md                                \
+    *.png                               \
+    *.po                                \
+    *.pot                               \
+    *.py                                \
+    *.qhcp                              \
+    *.qhp                               \
+    *.rej                               \
```
Comment (cjmayo) [inline MANIFEST.in:44]: Hopefully we will get 362 done!

File: MANIFEST.in (hunk 1 above)
Comment (mgedmin) [reply MANIFEST.in:13]: Oops.

File: MANIFEST.in (hunk 2 above)
Comment (mgedmin) [reply MANIFEST.in:44]: Why is there a *.rej pattern.  Why.

Oh!

**Comment by mgedmin** [comment]
> Just sdist warnings but picked up some interesting extensions:

Those were in the old MANIFEST.in, and probably indicate that there may be some additional build steps in the previous maintainer's release process that would generate files to be included in a release tarball.

> warning: no files found matching '\*.qhcp' under directory 'doc'
> warning: no files found matching '\*.qhp' under directory 'doc'

Google tells me these are Qt help files, probably deleted with the GUI code.

> warning: no files found matching '\*.bat' under directory 'tests'

Deleted in one of the cleanups maybe?

> warning: no files found matching '\*.cer' under directory 'tests'
> warning: no files found matching '\*.pfx' under directory 'tests'
> warning: no files found matching '\*.pvk' under directory 'tests'

`windows/makelccert.bat` has
```bat
:: File to generate a code signing certificate for LinkChecker
:: Needs makecert.exe and pkv2pfx.exe installed in the PATH
makecert -r -pe -n "CN=LinkChecker certificate" -b 01/01/2011 -e 01/01/2021 -eku 1.3.6.1.5.5.7.3.3 -sv linkchecker.pvk linkchecker.cer
pvk2pfx -pvk linkchecker.pvk -spc linkchecker.cer -pfx linkchecker.pfx
```

so these are some kind of SSL keys/certificates (pvk == private key?  WHY IS IT INCLUDED IN THE SDIST THAT SOUNDS LIKE A **TERRIBLE** IDEA???)

I checked the 9.3 tarball and it has some qhp/qhcp files, but no cer/pfx/pvk files.

**Pull request merged**